plugins/inputs/statsd/statsd.go

```diff
@@ -746,14 +751,16 @@ func (s *Statsd) aggregate(m metric) {
 		s.timings[m.hash] = cached
 	case "c":
 		// check if the measurement exists
-		_, ok := s.counters[m.hash]
+		counter, ok := s.counters[m.hash]
 		if !ok {
-			s.counters[m.hash] = cachedcounter{
+			counter = cachedcounter{
 				name:   m.name,
 				fields: make(map[string]interface{}),
 				tags:   m.tags,
 			}
 		}
+		counter.tm = tm
```

Comment:
Nitpick, but set `tm` inline for these:
```go
s.counters[m.hash] = cachedcounter{
  name:   m.name,
  fields: make(map[string]interface{}),
  tags:   m.tags,
  tm:     tm,      
}
```

Reply:
I originally did that, however I realized that would only set the time when the `cached*` structs were created. We need the time to be updated even if we don't have a new structure.

Reply:
ooh yes, good call